File: 后端代码/src/main/java/com/example/xyzmarket/controller/NotificationController.java
package com.example.xyzmarket.controller;

import com.example.xyzmarket.common.Result;
import com.example.xyzmarket.entity.Notification;
import com.example.xyzmarket.service.NotificationService;
import jakarta.servlet.http.HttpServletRequest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/notification")
public class NotificationController {

    @Autowired
    private NotificationService notificationService;

    /**
     * 获取我的通知列表
     */
    @GetMapping("/list")
    public Result<List<Notification>> getMyNotifications(HttpServletRequest request) {
        Long userId = (Long) request.getAttribute("userId");
        List<Notification> list = notificationService.getMyNotifications(userId);
        return Result.success(list);
    }

    /**
     * 获取未读通知数量
     */
    @GetMapping("/unread")
    public Result<Map<String, Long>> getUnreadCount(HttpServletRequest request) {
        Long userId = (Long) request.getAttribute("userId");
        long count = notificationService.getUnreadCount(userId);
        return Result.success(Map.of("count", count));
    }

    /**
     * 标记单条通知为已读
     */
    @PutMapping("/{id}/read")
    public Result<Void> markAsRead(@PathVariable Long id, HttpServletRequest request) {
        Long userId = (Long) request.getAttribute("userId");
        notificationService.markAsRead(id, userId);
        return Result.success();
    }

    /**
     * 标记所有通知为已读
     */
    @PutMapping("/read-all")
    public Result<Void> markAllAsRead(HttpServletRequest request) {
        Long userId = (Long) request.getAttribute("userId");
        notificationService.markAllAsRead(userId);
        return Result.success();
    }

}

File: 前端代码/项目/controllers/notification-controller.js
import { NotificationModel } from '../models/notification-model.js';

export class NotificationController {
  constructor() {
    this.notificationModel = new NotificationModel();
  }

  async getList() {
    try {
      return await this.notificationModel.getList();
    } catch (error) {
      throw error;
    }
  }

  async getUnreadCount() {
    try {
      return await this.notificationModel.getUnreadCount();
    } catch (error) {
      return 0;
    }
  }

  async markAsRead(id) {
    return this.notificationModel.markAsRead(id);
  }

  async markAllAsRead() {
    return this.notificationModel.markAllAsRead();
  }
}